File: EqualExperts/dotnet/FizzBuzz/Tracker.cs
using System;

namespace FizzBuzz
{
    public class Tracker
    {
        private int _numberCount = 0;
        private int _fizzCount = 0;
        private int _buzzCount = 0;
        private int _fizzBuzzCount = 0;
        private int _luckyCount = 0;

        public string GetTrackerReport() =>
            $"fizz: {_fizzCount} buzz: {_buzzCount} fizzbuzz: {_fizzBuzzCount} lucky: {_luckyCount} integer: {_numberCount}";

        public void TrackNumber() => _numberCount++;
        public void TrackFizz() => _fizzCount++;
        public void TrackBuzz() => _buzzCount++;
        public void TrackFizzBuzz() => _fizzBuzzCount++;
        public void TrackLucky() => _luckyCount++;
    }
}

File: EqualExperts/node/src/tracker.js
const counters = {
  lucky: 0,
  fizzbuzz: 0,
  fizz: 0,
  buzz: 0,
  integer: 0
}

const reset = () => {
  counters.lucky = 0
  counters.fizzbuzz = 0
  counters.fizz = 0
  counters.buzz = 0
  counters.integer = 0
}

const tracker = (value) => {
  switch (value) {
    case "lucky":
    case "fizzbuzz":
    case "fizz":
    case "buzz":
      counters[value]++
      break;

    default:
      counters.integer++
  }
}

const getReport = () => {
  return [
    `fizz: ${counters.fizz}`,
    `buzz: ${counters.buzz}`,
    `fizzbuzz: ${counters.fizzbuzz}`,
    `lucky: ${counters.lucky}`,
    `integer: ${counters.integer}`
  ].join(" ")
}

module.exports = {
  reset,
  tracker,
  getReport
}